sample/trace_key_test.go

```diff
@@ -160,3 +162,341 @@ func createTestTrace(t *testing.T, spans []testSpan) *types.Trace {
+						}
+					}
+				}
+
+				span := &types.Span{
+					Event: types.Event{
+						Data: spanData,
+					},
+				}
+
+				// Set as root span if it's the first one and we need root fields
+				if i == 0 && scenario.rootFields {
+					span.IsRoot = true
+					trace.RootSpan = span
+
+					// Add root-specific fields
+					for _, field := range scenario.fields {
+						if strings.HasPrefix(field, "root.") {
+							actualField := field[len("root."):]
+							span.Data[actualField] = fmt.Sprintf("root-%s-value", actualField)
+						}
+					}
+				}
+
+				trace.AddSpan(span)
+			}
+
+			b.ResetTimer()
+
+			for i := 0; i < b.N; i++ {
+				key, _ := generator.build(trace)
+				// Force evaluation to prevent the compiler from optimizing away the call
+				if len(key) == 0 {
+					b.Fatal("Empty key returned")
+				}
+			}
+		})
+	}
+}
+
+func TestDistinctValue_ValueTypeConsistency(t *testing.T) {
+	tests := []struct {
+		name           string
+		equalValues    []interface{} // These should all be considered identical
+		differentValue interface{}   // This should be considered a new unique value
+		expectedStr    string        // The expected string representation in Values()
+	}{
+		{
+			name:           "integer_and_string_2",
+			equalValues:    []interface{}{2, "2", 2.0},
+			differentValue: "2a",
+			expectedStr:    "2",
+		},
+		{
+			name:           "boolean_true",
+			equalValues:    []interface{}{true, "true"},
+			differentValue: 1,
+			expectedStr:    "true",
+		},
+		{
+			name:           "boolean_false",
+			equalValues:    []interface{}{false, "false"},
+			differentValue: 0,
+			expectedStr:    "false",
+		},
+		{
+			name:           "floating_point",
+			equalValues:    []interface{}{3.14, "3.14"},
+			differentValue: 3,
+			expectedStr:    "3.14",
+		},
+		{
+			name:           "zero_values",
+			equalValues:    []interface{}{0, "0", 0.0},
+			differentValue: false,
+			expectedStr:    "0",
+		},
+		{
+			name:           "negative_numbers",
+			equalValues:    []interface{}{-42, "-42", -42.0},
+			differentValue: "-42.0",
+			expectedStr:    "-42",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			dv := &distinctValue{
+				buf:              make([]byte, 0, 1024),
+				fields:           []string{"testField"},
+				values:           []map[uint64]string{make(map[uint64]string)},
+				maxDistinctValue: 100,
+			}
+
+			firstAdded := dv.AddAsString(tt.equalValues[0], 0)
+			require.True(t, firstAdded, "First value should be added successfully")
+			require.Equal(t, 1, dv.totalUniqueCount)
+
+			// All equal values should be considered duplicates
+			for i := 1; i < len(tt.equalValues); i++ {
+				added := dv.AddAsString(tt.equalValues[i], 0)
+				require.False(t, added,
+					"Value %v should be considered a duplicate of %v",
+					tt.equalValues[i], tt.equalValues[0])
+				require.Equal(t, 1, dv.totalUniqueCount,
+					"Count should remain 1 after adding duplicate %v", tt.equalValues[i])
```

Comment:
nit: this formatting is a mess. You have two choices: all one line, or all separate lines.
```
foo(
  a,
  b,
)
```